yt_dlp/extractor/nrk.py

```diff
@@ -284,11 +318,42 @@ def call_playback_api(item, query=None):
 
         return info
 
+    def _perform_login(self, username, password):
+        try:
+            self._download_json(
+                self._LOGIN_URL, None, headers={'Content-Type': 'application/json; charset=UTF-8', 'accept': 'application/json; charset=utf-8'},
+                data=json.dumps({
+                    'clientId': '',
+                    'hashedPassword': {'current': {
+                        'hash': password,
+                        'recipe': {
+                            'algorithm': 'cleartext',
+                            'salt': '',
+                        },
+                    },
+                    },
+                    'password': password,
+                    'username': username,
+                }).encode())
+
+            self._download_webpage('https://tv.nrk.no/auth/web/login/opsession', None)
+            response = self._download_json('https://tv.nrk.no/auth/session/tokenforsub/_', None)
+            self._AUTH_TOKEN = try_get(response, lambda x: x['session']['accessToken'])
+            self._API_CALL_HEADERS['authorization'] = f'Bearer {self._AUTH_TOKEN}'
+        except ExtractorError as e:
+            message = None
+            if isinstance(e.cause, HTTPError) and e.cause.status in (401, 400):
+                resp = self._parse_json(
+                    e.cause.response.read().decode(), None, fatal=False) or {}
+                message = next((error['message'] for error in resp['errors'] if error['field'] == 'Password'), None)
+            self.report_warning(message or 'Unable to log in')
+
 
 class NRKTVIE(InfoExtractor):
     IE_DESC = 'NRK TV and NRK Radio'
     _EPISODE_RE = r'(?P<id>[a-zA-Z]{4}\d{8})'
     _VALID_URL = rf'https?://(?:tv|radio)\.nrk(?:super)?\.no/(?:[^/]+/)*{_EPISODE_RE}'
+    _NETRC_MACHINE = 'nrk'
```

Comment:
I didn't include it originally, but adding it in both `NRKBaseIE` and `NRKTVIE` was the only way I found to silence the `Login with password is not supported for this website`-warning. Suggestions for better ways to deal with this issue would be much appreciated.

I'm actually not 100% sure the `_perform_login` method works in the EU outside of Norway though. It's difficult to test since being logged in or not makes no difference when requesting the hls streams from a Norwegian IP, and all the VPNs I've tried are blocked. I asked on the related issue if anyone was able to help with testing, but have yet hear back from anyone. However, it does seem to be able to fetch the auth token, so I assume it should work.

Reply:
I would just subclass from `NRKBaseIE` and remove it here